packages/gatsby-theme-docs/src/components/link.spec.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import {
+  createMemorySource,
+  createHistory,
+  LocationProvider,
+} from '@reach/router';
+import { useSiteData } from '../hooks/use-site-data';
+import Link from './link';
+
+jest.mock('../hooks/use-site-data');
+
+// for some types of tests you want a memory source
+const renderLink = (ui, initialPath) => {
+  const source = createMemorySource(initialPath);
+  const history = createHistory(source);
+  return render(<LocationProvider history={history}>{ui}</LocationProvider>);
+};
+
+const pathPrefix = '/prefix';
+const withPrefix = path => [pathPrefix, path].join('/').replace(/\/\/+/g, '/');
+
+describe('rendering', () => {
+  beforeEach(() => {
+    useSiteData.mockClear();
+    useSiteData.mockReturnValue({ pathPrefix });
+  });
+  const scenarios = [
+    {
+      title: 'image link',
+      props: { href: '/static/123' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'image-link' },
+    },
+    {
+      title: 'external link',
+      props: {
+        href: 'https://github.com/commercetools/commercetools-docs-kit',
+      },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'external-link' },
+    },
+    {
+      title: 'anchor link',
+      props: { href: '#title-1' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'anchor-link' },
+    },
+    {
+      title: 'anchor link to same page',
```

Comment:
Not sure about the naming convention that has the name of the expected behavior ("anchor") as the main keyword, but it's not a blocker.